### PR TITLE
[RNBW-3529] Wrong Value Shown After Currency Switch: UniqueTokenExpandedState

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
We are now updating `priceOfEth` when `UniqueTokenExpandedState` detects a change in `nativeCurrency`

## Dev checklist for QA: what to test
Generally ensure that the floor price and last sale price appropriately reflect the chosen native currency in `UniqueTokenExpandedState`.

To reproduce the issue in production:
* open up `UniqueTokenExpandedState` by tapping on an NFT (ensure the NFT you use has a floor price)
* tap on the floor price to toggle display from ETH to native currency (do not tap twice, you must leave the native currency displaying)
* close the modal and navigate to setting to change your selected native currency
* return to `UniqueTokenExpandedState`, when tapping on the floor price you should see the appropriate symbol but a value calculated based on your last native currency selection.

You may need to do the above twice to see the issue, but I never had to do more than that.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Review by brunobar79 — APPROVED** [review]
🤝

**Review by osdnk — APPROVED** [review]

File: src/components/expanded-state/UniqueTokenExpandedState.tsx [hunk 1]
```diff
@@ -208,6 +208,9 @@ const UniqueTokenExpandedState = ({
   } = useShowcaseTokens();
 
   const [floorPrice, setFloorPrice] = useState<string | null>(null);
+  const [priceOfEth, setPriceOfEth] = useState<number>(
+    ethereumUtils.getEthPriceUnit()
+  );
```
Comment (terrysahaidak) [inline src/components/expanded-state/UniqueTokenExpandedState.tsx:213]: I guess it should be something like this so we don't run this thing every rerender.

```suggestion
  const [priceOfEth, setPriceOfEth] = useState<number>(
    () => ethereumUtils.getEthPriceUnit()
  );
```

Comment (derHowie) [reply src/components/expanded-state/UniqueTokenExpandedState.tsx:213]: 🙇

**Review by terrysahaidak — APPROVED** [review]

**Comment by estebanmino** [comment]
@derHowie we got some conflicts, after that is this ready to qa? cc @ibrahimtaveras00

**Comment by derHowie** [comment]
This bug no longer exists in develop; closing.

**Pull request closed without merging**